front-end-components/src/components/charts/stat/component.tsx

```diff
@@ -0,0 +1,77 @@
+import { StatProps } from "src/components/charts/stat";
+import {
+  ChartDataSeries,
+  ChartSeriesValue,
+  ChartSeriesValueUnit,
+} from "src/components";
+import { useMemo } from "react";
+import classNames from "classnames";
+
+export function Stat<TData extends ChartDataSeries>(props: StatProps<TData>) {
+  const {
+    chartName,
+    className,
+    data,
+    displayIndex,
+    seriesLabelField,
+    valueField,
+    valueUnit,
+  } = props;
+
+  const entry = useMemo(() => {
+    const entry = Object.entries(data)[displayIndex];
+    if (!entry) {
+      return null;
+    }
+
+    return { label: entry[1][seriesLabelField], value: entry[1][valueField] };
+  }, [data, displayIndex, seriesLabelField, valueField]);
+
+  if (!entry) {
```

Comment:
This throw me for a moment - thinking is was rogue. Would it be better made this as a ternary on the main return?

Reply:
Could do, but in React I find it much more readable just to bomb out as soon as you can with `null` if there is nothing left to render

Reply:
Happy to add in a comment instead?

Reply:
Yeah, comment is fine.